core/app.js

```diff
@@ -96,6 +96,8 @@ app.use(_.get("/signin", function *(){
   var obj= yield auth.cas_login(ticket, SERVICE);
   this.session.user=obj.username;
   this.session.attributes=obj.attributes;
+  var client = db();
+  client.query("INSERT INTO users(netid) VALUES ($1);", [obj.username] )

```

Comment:
Can you add an `IF NOT EXISTS` here? I'm afraid this will throw an error if the user has already logged in once and it tries to insert another user.